packages/react-components/react-skeleton/SPEC.md

```diff
@@ -0,0 +1,101 @@
+# @fluentui/react-skeleton Spec
+
+## Background
+
+The `Skeleton` component is a temporary animation placeholder for when a service call takes time to return data and we don't want to block rendering the rest of the UI.
+
+## Prior Art
+
+### Open UI
+
+| Library    | Component Name | Spec Link                                                                       | Notes                                                                                                                                                                  |
+| ---------- | -------------- | ------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Ant Design | Skeleton       | [Skeleton](https://ant.design/components/skeleton/)                             | Has an `avatar`, `button`, `list`, and `togglebutton`. Has animated and non-animated versions, and shape variants for the component skeletons(rounded or rectangular). |
+| Fast       | Skeleton       | [Skeleton](https://explore.fast.design/components/fast-skeleton)                | Has three components: `Avatar`, `List`, and `Button`. No shape variants.                                                                                               |
+| UI Fabric  | Shimmer        | [Shimmer](https://developer.microsoft.com/en-us/fluentui#/controls/web/shimmer) | Shimmer has a way to pass in custom elements as well as the defaults. There is also a prop to change the colors of the Skeleton components.                            |
+
+### Comparison of v8 and v0
+
+The existing components are:
+
+- v8
+  - `Shimmer`
+- v0
+
+## Sample Code
+
+Basic example:
+
+```jsx
+import { Skeleton, SkeletonItem } from '@fluentui/react-skeleton';
+
+function App() {
+  return (
+    <Skeleton>
+      <SkeletonItem />
+    </Skeleton>
+  );
+}
+```
+
+## Variants
+
+- There are three basic `Skeleton` shapes
+  - `Rectangle`
+  - `Circle`
+  - `Square`
```

Comment:
What's the advantage of having square when we already have rectangle?

Reply:
The idea is to allow the users to specify one `size` prop and one `shape` prop and have it render the shape without having to specify `height` and `width`

Reply:
In that case how does a user specify the dimensions of a rectangle?

Reply:
For the case of a rectangle, the default width is set to `auto` and users can pass a value to the `size` prop, which will set the height of the rectangle. If they want a specific width, they can add some styling to set the width.